### PR TITLE
Add support for the `any` builtin (#54)

### PR DESCRIPTION
**Pull request by meadori**
Implement the `any` as defined here:

    * https://docs.python.org/2/library/functions.html#any

### CODE REVIEW

**Comment by choleraehyq** [comment]
Haha, I'm about to send a similar PR.
LGTM.

File: runtime/builtin_types.go [hunk 1]
```diff
@@ -637,3 +627,31 @@ func numberToBase(prefix string, base int, o *Object) string {
...
 	return prefix + s
 }
+
+// quantifyIterable implements the builtins "all" and "any".
+// It takes a condition function, "cond", that determines when
+// the iteration should stop.  If the iteration stops before
+// all the items are seen, then "condTrue" is returned. Otherwise,
+// "condFalse" is returned. For more details see:
+// https://docs.python.org/2/library/functions.html#all
+func quantifyIterable(f *Frame, o *Object, cond func(bool) bool, condTrue *Object, condFalse *Object) (*Object, *BaseException) {
```
Comment (trotterdylan) [inline runtime/builtin_types.go]: How about something like this:

```
func seqFindFirst(f *Frame, iterable *Object, pred func(*Object) bool) (bool, *BaseException)
```

Then you can define predicates `seqFindFirstTrue()` and `seqFindFirstFalse()` to be used by `any` and `all`. `seqFindFirst()` can then be reused by `[grumpy.Contains()](https://github.com/google/grumpy/blob/master/runtime/core.go#L71)`.

What do you think?

Comment (trotterdylan) [inline runtime/builtin_types.go]: Actually the predicate would have to be able to return an exception, so:

```
func seqFindFirst(f *Frame, iterable *Object, pred func(*Object) (bool, *BaseException)) (bool, *BaseException)
```

Comment (meadori) [inline runtime/builtin_types.go]: I like it!  Thanks for the suggestion.  I will have a new patch shortly.

**Comment by trotterdylan** [comment]
Thanks for doing this! LGTM

**Pull request merged**